todo_app/templates/index.html

```diff
@@ -63,34 +60,81 @@ <h5>
                   <input type="submit" name="complete" class="btn-secondary" value="✔️" data-source="{{item.id}}">
                   <input type="hidden" name="complete_id" value="{{item.id}}">
                 </form>
-                </ul>
               </li>
-                {% endfor %}  
+              {% endfor %}  
+            </ul>
 
       <h2 class="text-center">COMPLETED:</h2>
 
       <ul class="list-group">
-        {% for item in done_tasks %}
+        {% for item in view_model.done_items %}
           <li class="list-group-item">
             <h5>
+              {% if loop.index <= 5 %}
+
               {{item.name}}
               <form class="float-end" method="POST" action="/remove/{{item.id}}">
-              <input type="submit" name="remove" class="btn-secondary" value="🗑️">
-              <input type="hidden" name="remove_id" value="{{item.id}}">
-            </form>
-            <form class="float-end" method="POST" action="/mark_doing/{{item.id}}">
-              <input type="submit" name="doing" class="btn-secondary" value="doing" data-source="{{item.id}}">
-              <input type="hidden" name="doing_id" value="{{item.id}}">
-            </form>
-            <form class="float-end" method="POST" action="/mark_to_do/{{item.id}}">
-              <input type="submit" name="to_do" class="btn-secondary" value="to do" data-source="{{item.id}}">
-              <input type="hidden" name="incomplete_id" value="{{item.id}}">
-            </form>
-            </ul>
+                <input type="submit" name="remove" class="btn-secondary" value="🗑️">
+                <input type="hidden" name="remove_id" value="{{item.id}}">
+              </form>
+              <form class="float-end" method="POST" action="/mark_doing/{{item.id}}">
+                <input type="submit" name="doing" class="btn-secondary" value="doing" data-source="{{item.id}}">
+                <input type="hidden" name="doing_id" value="{{item.id}}">
+              </form>
+              <form class="float-end" method="POST" action="/mark_to_do/{{item.id}}">
+                <input type="submit" name="to_do" class="btn-secondary" value="to do" data-source="{{item.id}}">
+                <input type="hidden" name="incomplete_id" value="{{item.id}}">
+              </form>
+            
+            
+            </li>
+          
+              {% endif %}  
+              {% endfor %}  
+    
+            
+               
+          <details>
+          <summary><u>Show older completed tasks</u></summary>
 
-           
-          </li>
-          {% endfor %}  
+          {% for item in view_model.done_items %}
+          
+
+          {% if loop.index > 5 %}
```

Comment:
Nice work on this task, I can see the structure works but note that this logic assumes that the tasks will come through in the right order (i.e. newest first); but from my testing I'm not convinced that they do! If I have 5 items completed and add a 6th, I can only see the oldest items by default, and the newest is the one dropped into this extra space

Reply:
I am aware. I was trying to use the due-by date option, but didn't have time and left it like that.